### PR TITLE
remove kvstore::ResultCode

### PR DESCRIPTION
**Pull request by panda-sheep**
As title!
reply on https://github.com/vesoft-inc/nebula-common/pull/513

### CODE REVIEW

**Comment by panda-sheep** [comment]
Ready to review

File: src/storage/BaseProcessor.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 namespace nebula {
 namespace storage {
 
-using PartCode = std::pair<PartitionID, kvstore::ResultCode>;
+using PartCode = std::pair<PartitionID, nebula::cpp2::ErrorCode>;
```
Comment (critical27) [inline src/storage/BaseProcessor.h:25]: Could delete it.

File: src/storage/BaseProcessor.h [hunk 2]
```diff
@@ -89,28 +90,24 @@ class BaseProcessor {
...
                                             nebula::meta::cpp2::PropertyType type);
 
-    void pushResultCode(cpp2::ErrorCode code, PartitionID partId);
+    void pushResultCode(nebula::cpp2::ErrorCode code, PartitionID partId);
 
-    void pushResultCode(cpp2::ErrorCode code, PartitionID partId, HostAddr leader);
+    void pushResultCode(nebula::cpp2::ErrorCode code, PartitionID partId, HostAddr leader);
 
-    void handleErrorCode(kvstore::ResultCode code, GraphSpaceID spaceId, PartitionID partId);
+    void handleErrorCode(nebula::cpp2::ErrorCode code,
```
Comment (critical27) [inline src/storage/BaseProcessor.h:102]: `pushResultCode` and `handleErrorCode` should only save one now, they are for sync method.

File: src/storage/BaseProcessor.h [hunk 3]
```diff
@@ -89,28 +90,24 @@ class BaseProcessor {
...
+                         GraphSpaceID spaceId,
+                         PartitionID partId);
 
     void handleLeaderChanged(GraphSpaceID spaceId, PartitionID partId);
 
     void handleAsync(GraphSpaceID spaceId,
                      PartitionID partId,
-                     kvstore::ResultCode code);
```
Comment (critical27) [inline src/storage/BaseProcessor.h:109]: 👍

File: src/storage/BaseProcessor.h [hunk 4]
```diff
@@ -89,28 +90,24 @@ class BaseProcessor {
...
+    nebula::cpp2::ErrorCode writeResultTo(WriteResult code, bool isEdge);
 
     nebula::meta::cpp2::ColumnDef columnDef(std::string name,
                                             nebula::meta::cpp2::PropertyType type);
 
-    void pushResultCode(cpp2::ErrorCode code, PartitionID partId);
+    void pushResultCode(nebula::cpp2::ErrorCode code, PartitionID partId);
 
-    void pushResultCode(cpp2::ErrorCode code, PartitionID partId, HostAddr leader);
+    void pushResultCode(nebula::cpp2::ErrorCode code, PartitionID partId, HostAddr leader);
```
Comment (critical27) [inline src/storage/BaseProcessor.h:100]: Check if we need this one with extra `HostAddr leader`, if not in used, could delete it as well.

File: src/storage/CommonUtils.cpp [hunk 5]
```diff
@@ -10,83 +10,19 @@
 namespace nebula {
 namespace storage {
 
-cpp2::ErrorCode CommonUtils::to(const Status& status) {
+nebula::cpp2::ErrorCode CommonUtils::to(const Status& status) {
```
Comment (critical27) [inline src/storage/CommonUtils.cpp:13]: This `to` could be removed

**Review by critical27 — COMMENTED** [review]
Gooooood job, there are some extra work to do, see the comments. Just clean everything redundant.

File: src/kvstore/Common.h [hunk 6]
```diff
@@ -12,45 +12,12 @@
...
-    ERR_BACKUP_EMPTY_TABLE  = -26,
-    ERR_DATA_CONFLICT_ERROR = -27,
-    ERR_USER_CANCELLED      = -98,
-    ERR_PARTIAL_RESULT      = -99,
-    ERR_UNKNOWN             = -100,
-};
-
-inline std::ostream& operator <<(std::ostream& os, const ResultCode& rc) {
+inline std::ostream& operator <<(std::ostream& os, const nebula::cpp2::ErrorCode& rc) {
     return os << static_cast<int32_t>(rc);
```
[the remark this answers is not in the document]
Comment (panda-sheep) [reply src/kvstore/Common.h:21]: Yeah.

File: src/storage/BaseProcessor.h (hunk 4 above)
Comment (panda-sheep) [reply src/storage/BaseProcessor.h:100]: handleLeaderChanged  need this function. 
 I added a default value to the `leader`, and then removed `pushResultCode `without` HostAddr leader`

File: src/storage/BaseProcessor.h (hunk 2 above)
Comment (panda-sheep) [reply src/storage/BaseProcessor.h:102]: `handleErrorCode `contains `handleLeaderChanged `and `pushResultCode`.
I think it is clearer to separate handleLeaderChanged and pushResultCode.
Of course they can also be merged together.

File: src/storage/CommonUtils.cpp (hunk 5 above)
Comment (panda-sheep) [reply src/storage/CommonUtils.cpp:13]: Good

**Comment by panda-sheep** [comment]
Ready to review again.

**Review by critical27 — APPROVED** [review]
Generally LGTM now! Well done, huge PR.

**Comment by critical27** [comment]
Wait https://github.com/vesoft-inc/nebula-common/pull/513

**Comment by panda-sheep** [comment]
fix rebase master build failed

**Review by liuyu85cn — APPROVED** [review]

**Review by critical27 — APPROVED** [review]
Good job

**Pull request merged**